src/mpid/ch4/generic/am/mpidig_am_recv.h

```diff
@@ -158,6 +153,30 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_do_irecv(void *buf, MPI_Aint count, MPI_Data
             mpi_errno = MPIDIG_do_long_ack(unexp_req);
             MPIR_ERR_CHECK(mpi_errno);
             goto fn_exit;
+        } else if (MPIDIG_REQUEST(unexp_req, req->status) & MPIDIG_REQ_RTS) {
+            /* Matching receive is now posted, tell the netmod/shmmod */
+            MPIR_Datatype_add_ref_if_not_builtin(datatype);
+            MPIDIG_REQUEST(unexp_req, datatype) = datatype;
+            MPIDIG_REQUEST(unexp_req, buffer) = (char *) buf;
+            MPIDIG_REQUEST(unexp_req, count) = count;
+            if (*request == NULL) {
+                /* Regular (non-enqueuing) path: MPIDIG is responsbile for allocating
+                 * a request. Here we simply return `unexp_req` */
+                *request = unexp_req;
+                /* Mark `match_req` as NULL so that we know nothing else to complete when
+                 * `unexp_req` finally completes. (See below) */
+                MPIDIG_REQUEST(unexp_req, req->rreq.match_req) = NULL;
+            } else {
+                /* Enqueuing path: CH4 already allocated a request.
+                 * Record the passed `*request` to `match_req` so that we can complete it
+                 * later when `unexp_req` completes.
+                 * See MPIDI_recv_target_cmpl_cb for actual completion handler. */
+                MPIDIG_REQUEST(unexp_req, req->rreq.match_req) = *request;
+            }
+            MPIDIG_REQUEST(unexp_req, req->status) &= ~MPIDIG_REQ_UNEXPECTED;
+            mpi_errno = MPIDIG_do_cts(unexp_req);
+            MPIR_ERR_CHECK(mpi_errno);
+            goto fn_exit;
```

Comment:
Since we are not doing RTS in this PR, this branch is not needed.

Reply:
See the comments about `MPIDIG_SEND_LONG_REQ` below.